test/shopify-cli/task/update_dashboard_urls_test.rb

```diff
@@ -0,0 +1,188 @@
+require 'test_helper'
+
+module ShopifyCli
+  module Tasks
+    class UpdateDashboardURLSTest < MiniTest::Test
+      include TestHelpers::Partners
+
+      def test_url_is_not_transformed_if_same
```

Comment:
I think it's worth adding one more test case: that the whitelist URLs are updated if the app URL is already updated. Like if someone accidentally saved a stale state in the Dashboard somehow.